packages/flutter_bloc/lib/src/bloc_builder.dart

```diff
@@ -15,62 +15,65 @@ typedef BlocWidgetBuilder<S> = Widget Function(BuildContext context, S state);
 /// [BlocBuilder] with the current state.
 typedef BlocBuilderCondition<S> = bool Function(S previous, S current);
 
+/// {@template blocbuilder}
+/// [BlocBuilder] handles building a widget in response to new states.
+/// [BlocBuilder] is analogous to [StreamBuilder] but has simplified API
+/// to reduce the amount of boilerplate code needed as well as bloc-specific performance improvements.
+
+/// Please refer to [BlocListener] if you want to "do" anything in response to state changes such as
+/// navigation, showing a dialog, etc...
+///
+/// If the bloc parameter is omitted, [BlocBuilder] will automatically perform a lookup using
+/// [BlocProvider] and the current [BuildContext].
+///
+/// ```dart
+/// BlocBuilder<BlocA, BlocAState>(
+///   builder: (context, state) {
+///   // return widget here based on BlocA's state
+///   }
+/// )
+/// ```
+///
+/// Only specify the bloc if you wish to provide a bloc that is otherwise
+/// not accessible via [BlocProvider] and the current [BuildContext].
+///
+/// ```dart
+/// BlocBuilder<BlocA, BlocAState>(
+///   bloc: blocA,
+///   builder: (context, state) {
+///   // return widget here based on BlocA's state
+///   }
+/// )
+/// ```
+///
+/// An optional `condition` can be implemented for more granular control
+/// over how often [BlocBuilder] rebuilds.
+/// The `condition` function will be invoked on each bloc state change.
```

Comment:
we can make `[condition]` etc. for fields like this

Reply:
The same for all listener, builder etc. For all fields essentially.

Reply:
Do you know what impact this will have on the generated docs? I'll run dart-doc and test it out

Reply:
I think it will (only) allow to create more links between classes/methods/fields so user can navigate within the implementation in API reference page (https://pub.dev/documentation/bloc/latest/).

Reply:
Basically, I think you should want to make a reference from everything you can, or that is how Flutter does it. And then the description is not just a plain text, but it has a context/references.

Check for example https://api.flutter.dev/flutter/widgets/Column-class.html

Dunno, what do you think?

Reply:
It seems you can reference only methods, it's arguments and fields. But thats probably ok.

Reply:
Yeah I just tested it locally and it was able to create references 👍 I just updated the PR can you take a look?